src/promptflow/promptflow/batch/_result.py

```diff
@@ -0,0 +1,144 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from dataclasses import dataclass
+from datetime import datetime
+from itertools import chain
+from typing import Any, Dict, List, Mapping
+
+from promptflow._utils.exception_utils import RootErrorCode
+from promptflow._utils.openai_metrics_calculator import OpenAIMetricsCalculator
+from promptflow.contracts.run_info import Status
+from promptflow.executor._result import AggregationResult, LineResult
+
+
+@dataclass
+class LineError:
+    line_number: int
+    error: Dict[str, Any]
```

Comment:
Add more comment to explain what is in the dict.

Also I guess it should a mapping? (immutable)